graylog2-server/src/main/java/org/graylog2/periodical/ESVersionCheckPeriodical.java

```diff
@@ -0,0 +1,125 @@
+import org.graylog2.storage.ElasticsearchVersion;
+import org.graylog2.storage.versionprobe.VersionProbe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+public class ESVersionCheckPeriodical extends Periodical {
+    private static final Logger LOG = LoggerFactory.getLogger(ESVersionCheckPeriodical.class);
+    private final Version initialElasticsearchVersion;
+    private final Optional<Version> versionOverride;
+    private final List<URI> elasticsearchHosts;
+    private final VersionProbe versionProbe;
+    private final NotificationService notificationService;
+
+    @Inject
+    public ESVersionCheckPeriodical(@ElasticsearchVersion Version elasticsearchVersion,
+                                    @Named("elasticsearch_version") @Nullable Version versionOverride,
+                                    @Named("elasticsearch_hosts") List<URI> elasticsearchHosts,
+                                    VersionProbe versionProbe,
+                                    NotificationService notificationService) {
+        this.initialElasticsearchVersion = elasticsearchVersion;
+        this.versionOverride = Optional.ofNullable(versionOverride);
+        this.elasticsearchHosts = elasticsearchHosts;
+        this.versionProbe = versionProbe;
+        this.notificationService = notificationService;
+    }
+
+    @Override
+    public boolean runsForever() {
+        return false;
+    }
+
+    @Override
+    public boolean stopOnGracefulShutdown() {
+        return true;
+    }
+
+    @Override
+    public boolean masterOnly() {
+        return true;
+    }
+
+    @Override
+    public boolean startOnThisNode() {
+        return true;
+    }
+
+    @Override
+    public boolean isDaemon() {
+        return false;
+    }
+
+    @Override
+    public int getInitialDelaySeconds() {
+        return 0;
+    }
+
+    @Override
+    public int getPeriodSeconds() {
+        return 30;
+    }
+
+    @Override
+    protected Logger getLogger() {
+        return LOG;
+    }
+
+    @Override
+    public void doRun() {
+        if (versionOverride.isPresent()) {
+            LOG.debug("Elasticsearch version is set manually. Not running check.");
+            return;
+        }
+
+        final Optional<Version> probedVersion = this.versionProbe.probe(this.elasticsearchHosts);
+
+        probedVersion.ifPresent(version -> {
+            if (compatible(this.initialElasticsearchVersion, version)) {
+                notificationService.fixed(Notification.Type.ES_VERSION_MISMATCH);
+            } else {
+                LOG.warn("Elasticsearch version currently running (" + version.toString() + ") is incompatible with the " +
+                        "one Graylog was started with (" + initialElasticsearchVersion.toString() + ")" +
+                        " - a restart is required!");
```

Comment:
Tiny nitpick: use placeholder instead of manual string concat makes it a bit more readable. :smile: 
```suggestion
                LOG.warn("Elasticsearch version currently running ({}) is incompatible with the " +
                        "one Graylog was started with ({}) - a restart is required!", version, initialElasticsearchVersion);
```

Reply:
Indeed! :)